File: 3/python/euler3.py
#! /usr/bin/python
import math
# WIP - Not done with this one
def is_prime(n):
	i = 2
	half_n = int(math.floor(n / 2))
	while i <= half_n:
		if n % i == 0:
			return False
		i += 1
	return True

def is_prime_fast(n):
	if n % 2 == 0 or n % 3 == 0:
		return False

	sqrt_n = int(math.ceil(math.sqrt(n)))
	i = 5
	while(i < sqrt_n):
		if n % i == 0 or n % (i + 2) == 0:
			return False
		i += 6

	return True

up_to = 600851475143
largest_prime = 0
for i in range(int(math.ceil(up_to / 2)), 1, -1):
	if(up_to % i == 0 and is_prime_fast(i)):
		largest_prime = i
		print('Found')
		break;

print(largest_prime)

File: 3/c++/euler3.cpp
#include <iostream>
#include <math.h>

using namespace std;

bool is_prime(int n) {
  if(n % 2 == 0 || n % 3 == 0)
		return false;

  for(int i = 5; i <= sqrt(n); i += 6) {
    if(n % i == 0 || n % (i + 2) == 0)
      return false;
  }
  return true;
}

int main() {
  long up_to = 600851475143;
  int largest_prime = 0;

  for(int i = sqrt(up_to); i > 1; i--) {
    if(is_prime(i) && up_to % i == 0) {
      largest_prime = i;
      break;
    }
  }

  cout << largest_prime << endl;
}